File: c/sorting/selectionSort.c
#include <stdio.h>
void swap(int *a, int *b)
{
    int temp;
    temp = *a;
    *a = *b;
    *b = temp;
}
void selectionSort(int arr[], int n)
{
    int least, p, i, j;
    for (i = 0; i <=n-1; i++)
    {
        least = arr[i];
        p = i;
        for (j = i + 1; j <= n-1; j++)
        {
            if (arr[j] < least)
            { // Compare with 'least' instead of 'arr[i]'
                least = arr[j];
                p = j;
            }
            
        }
int temp;
temp=arr[i];
arr[i]=arr[p];
arr[p]=temp;
        // swap(&arr[i], &arr[p]);
        int k=0;
            for (k=0; k < n; k++)
            {
                printf("%d\t", arr[k]);
            }
            printf("\n");
    }
}

int main()
{
    int i, n;
    int arr[] = {6, 2, 9, 3, 100, 32};
    n = sizeof(arr) / sizeof(int);
    selectionSort(arr, n);
    for (i = 0; i < n; i++)
    {
        printf("%d\t", arr[i]);
    }
}

File: Javascript/Sorting/selectionSort.js
function selectionSort(arr) {
    let n = arr.length;
    // One by one move boundary of unsorted subarray
    for (let i = 0; i < n-1; i++) {
        // Find the minimum element in unsorted array
        let min_idx = i;
        for (let j = i+1; j < n; j++) {
            if (arr[j] < arr[min_idx]) {
                min_idx = j;
            }
        }
        // Swap the found minimum element with the first element of unsorted array
        let temp = arr[min_idx];
        arr[min_idx] = arr[i];
        arr[i] = temp;
    }
    return arr;
}

function main() {
    let my_list=[5,8,7,4,2,9,12,13,11];
    // Make a copy of the original unsorted list
    let unsorted_list = [...my_list];

    let sorted_list = selectionSort(my_list);
    console.log("UnSorted list/Array", unsorted_list);
    console.log("Sorted list/Array  ", sorted_list);
}

main();
